clang/test/SemaSYCL/implicit_kernel_type.cpp

```diff
@@ -23,10 +23,10 @@ int main() {
   queue q;
 
 #if defined(WARN)
-  // expected-error@#KernelSingleTask {{'InvalidKernelName1' should be globally visible}}
+  // expected-error@#KernelSingleTask {{non-forward-declarable type 'InvalidKernelName1' is invalid; provide a forward declarable kernel name at namespace scope}}
   // expected-note@+7 {{in instantiation of function template specialization}}
 #elif defined(ERROR)
-  // expected-error@#KernelSingleTask {{'InvalidKernelName1' should be globally visible}}
+  // expected-error@#KernelSingleTask {{non-forward-declarable type 'InvalidKernelName1' is invalid; provide a forward declarable kernel name at namespace scope}}
   // expected-note@+4 {{in instantiation of function template specialization}}
```

Comment:
This is not about your patch but the test. It seems WARN and ERROR macros were defined to separate errors from warnings, however in this test case error is emitted in both cases. We probably can remove if condition for macro values here.

Reply:
Thanks @Fznamznon for noticing this. I have removed WARN macro because the test is generating error message only.